### ISSUE TITLE
[Feature Request] Show only fonts containing a given character (or series of characters)

### ISSUE THREAD

**Issue by pinnerup** [FontManager/font-manager#102]
Hiya

Thanks for an awsome font management tool. I'd like to request a feature that's easily available on other platforms, but apparently very hard to come by on Gnome systems.

Say I'm working with a non-Latin alphabet, and only a small fraction of my installed fonts contain this character range, I would like to be able to see which of my installed fonts contain a given character (or series of characters). 

Say I want to typeset the Greek word γρᾰφή – I should be able to input it somewhere and get a list of only the fonts that actually contain these characters (and not all the other fonts that do not). 

It could be implemented as an option under the "Browse" mode.

**Comment by JerryCasiano** [comment]
So...  It might be a while before I get around to improving our search. 

For now, starting a search with a colon will trigger a character search.

Please test with the latest commit and provide some feedback on whether those results match your needs/expectations.

**Comment by pinnerup** [comment]
Fantastic! Thank you very much. I've tested it, and this works out for me. This was a much faster response that I'd hoped for :)

**Comment by pinnerup** [comment]
Humm, it seems doing such a search (e.g. ":ϣ") now causes Font Manager to immediately terminate. At least on my computer (Ubuntu 20.10).

**Comment by JerryCasiano** [comment]
Unfortunately I can't reproduce this on a fresh up to date install of 20.10

![Screenshot from 2021-01-06 18-28-03](https://user-images.githubusercontent.com/11490124/103832828-51cc9d00-504d-11eb-9672-680e49c3f918.png)

Are there any warning or critical messages printed to the terminal if you run the application from the command line?

Is there anything different about your installation? Any customizations maybe?

If not, you'll need to install debug symbol packages and get a backtrace since I can't reproduce it in order to do that...

**Comment by pinnerup** [comment]
Thank you for the prompt reply :)

When I run `font-manager` from the command prompt and attempt the search for ":ϣ" (or indeed ":" followed by any letter or even just searching for ":"), I get this in the terminal:

```
pinnerup@ergi:~$ font-manager 
**
[font-manager]:ERROR:../lib/common/font-manager-fontconfig.c:143:is_legacy_format: assertion failed: (FcPatternGetString(pat, FC_FONTFORMAT, 0, &format) == FcResultMatch)
Bail out! [font-manager]:ERROR:../lib/common/font-manager-fontconfig.c:143:is_legacy_format: assertion failed: (FcPatternGetString(pat, FC_FONTFORMAT, 0, &format) == FcResultMatch)
Aborted (core dumped)
```

**Comment by JerryCasiano** [comment]
Well, that is somewhat helpful.

Are you using the PPA version or the upstream version?

**Comment by pinnerup** [comment]
Sorry, I should have thought of trying that out. The above was using version 0.7.8 that came with Ubuntu 20.10.

I've now installed 0.8.4 from the PPA, and the problem has disappeared. Thanks for the help :)

**Comment by JerryCasiano** [comment]
Of course. Your welcome.

This was fixed a few months ago in c46a030642244d4508053099d6f841b8318a13c6 and should be available in the regular repositories soon, I think it's in Testing at this point.

Anyway, closing this out for now.
